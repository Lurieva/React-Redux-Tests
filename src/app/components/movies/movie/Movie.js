import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import './movie.scss';

const Movie = ({ movie }) => {
    const year = new Date(movie.release_date).getFullYear();
    const genres = (movie.genres).join(' & ');

    return (
        <div className="movie">
            <div className="movie__content">
                <img className="movie__content__poster" 
                    src={movie.poster_path} 
                    alt={movie.title} />
                <div className="movie__content__info">
                    <Link className="movie__content__info__title" 
                          to={`/movies/${movie.id}`}>
                          {movie.title}
                    </Link>
                    <div className="movie__content__info__year">{year}</div>
                </div>
                <div className="movie__content__genres">{genres}</div>
            </div>
        </div>
    );
};

export default Movie;