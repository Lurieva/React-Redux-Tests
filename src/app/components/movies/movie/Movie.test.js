import React from 'react';
import { shallow } from 'enzyme';
import Movie from './Movie';

describe('Movie', () => {
    const movie = {
        id: 1,
        poster_path: 'path',
        title: 'title',
        release_date: '12-01-2018',
        genres: ['comedy']
    };

    it('should be defined', () => {
        expect(Movie).toBeDefined();
    });

    it('should render correctly ', () => {
        const tree = shallow(<Movie movie={movie} />);
        expect(tree).toMatchSnapshot();
    });
});