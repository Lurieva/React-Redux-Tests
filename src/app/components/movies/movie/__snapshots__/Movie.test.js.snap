// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should render correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    movie={
      Object {
        "genres": Array [
          "comedy",
        ],
        "id": 1,
        "poster_path": "path",
        "release_date": "12-01-2018",
        "title": "title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="movie__content"
      >
        <img
          alt="title"
          className="movie__content__poster"
          src="path"
        />
        <div
          className="movie__content__info"
        >
          <Link
            className="movie__content__info__title"
            replace={false}
            to="/movies/1"
          >
            title
          </Link>
          <div
            className="movie__content__info__year"
          >
            2018
          </div>
        </div>
        <div
          className="movie__content__genres"
        >
          comedy
        </div>
      </div>,
      "className": "movie",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="title"
            className="movie__content__poster"
            src="path"
          />,
          <div
            className="movie__content__info"
          >
            <Link
              className="movie__content__info__title"
              replace={false}
              to="/movies/1"
            >
              title
            </Link>
            <div
              className="movie__content__info__year"
            >
              2018
            </div>
          </div>,
          <div
            className="movie__content__genres"
          >
            comedy
          </div>,
        ],
        "className": "movie__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "title",
            "className": "movie__content__poster",
            "src": "path",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="movie__content__info__title"
                replace={false}
                to="/movies/1"
              >
                title
              </Link>,
              <div
                className="movie__content__info__year"
              >
                2018
              </div>,
            ],
            "className": "movie__content__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "title",
                "className": "movie__content__info__title",
                "replace": false,
                "to": "/movies/1",
              },
              "ref": null,
              "rendered": "title",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2018,
                "className": "movie__content__info__year",
              },
              "ref": null,
              "rendered": 2018,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "comedy",
            "className": "movie__content__genres",
          },
          "ref": null,
          "rendered": "comedy",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="movie__content"
        >
          <img
            alt="title"
            className="movie__content__poster"
            src="path"
          />
          <div
            className="movie__content__info"
          >
            <Link
              className="movie__content__info__title"
              replace={false}
              to="/movies/1"
            >
              title
            </Link>
            <div
              className="movie__content__info__year"
            >
              2018
            </div>
          </div>
          <div
            className="movie__content__genres"
          >
            comedy
          </div>
        </div>,
        "className": "movie",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="title"
              className="movie__content__poster"
              src="path"
            />,
            <div
              className="movie__content__info"
            >
              <Link
                className="movie__content__info__title"
                replace={false}
                to="/movies/1"
              >
                title
              </Link>
              <div
                className="movie__content__info__year"
              >
                2018
              </div>
            </div>,
            <div
              className="movie__content__genres"
            >
              comedy
            </div>,
          ],
          "className": "movie__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "title",
              "className": "movie__content__poster",
              "src": "path",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="movie__content__info__title"
                  replace={false}
                  to="/movies/1"
                >
                  title
                </Link>,
                <div
                  className="movie__content__info__year"
                >
                  2018
                </div>,
              ],
              "className": "movie__content__info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "title",
                  "className": "movie__content__info__title",
                  "replace": false,
                  "to": "/movies/1",
                },
                "ref": null,
                "rendered": "title",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 2018,
                  "className": "movie__content__info__year",
                },
                "ref": null,
                "rendered": 2018,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "comedy",
              "className": "movie__content__genres",
            },
            "ref": null,
            "rendered": "comedy",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
